p2p/client.go

```diff
@@ -313,6 +337,36 @@ func (c *Client) setupPeerDiscovery(ctx context.Context) error {
 	return nil
 }
 
+func (c *Client) setupBlockedPeers(peers []peer.AddrInfo) error {
+	if c.gater == nil {
+		return fmt.Errorf("connection gater is nil, cannot blacklist peers")
+	}
+	if len(peers) > 0 {
+		for _, p := range peers {
+			err := c.gater.BlockPeer(p.ID)
```

Comment:
recommend inlined error check for concise syntax. 
```
if err := c.gater.BlockPerr(p.ID); err != nil {
     return err
}
```
you only need to split the err check if you have a non-err return value and you intend to use it outside of the check. e.g.,
```
xyz, err := func_call()
err != nil {
}
use(xyz)
```
you can do the same throughout

Reply:
Very good point and I really like the explanation. In the entire codebase there are many places where I used the longer form without actually thinking about it - we'll need to improve this ;)

Reply:
I've got a macro for the long form but not the short one... that's why my code is full of the long ones!